libs/ml/include/ml/ops/divide.hpp

```diff
@@ -44,39 +45,81 @@ class Divide : public fetch::ml::Ops<T>
   virtual void Forward(VecTensorType const &inputs, ArrayType &output)
   {
     assert(inputs.size() == 2);
-    assert(inputs.at(0).get().shape() == inputs.at(1).get().shape());
     assert(inputs.at(0).get().shape() == output.shape());
 
-    fetch::math::Divide(inputs.at(0).get(), inputs.at(1).get(), output);
+    if ((inputs.at(0).get().shape() == inputs.at(1).get().shape()) ||
+        (inputs.at(1).get().size() > 1))
+    {  // array / array
```

Comment:
I think you actually only need to check that inputs.at(1).get().size() > 1
This forward Divide will just invoke Broadcast if it needs to and work just fine right?